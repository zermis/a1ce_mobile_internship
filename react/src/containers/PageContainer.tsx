import React from 'react';
import { View, StyleSheet, Dimensions, Platform } from 'react-native';
import Header from '../components/shared/Header';
import Body from '../components/shared/Body';

interface Props {
    isMain: boolean;
    headerContent?: JSX.Element;
    bodyContent?: JSX.Element;
    headerStyles?: object;
    bodyStyle?: object;
    accessibilityLabel?: string;
}

const { height, width } = Dimensions.get('window');

const PageContainer = (props: Props) => {
    const headerStyle = props.isMain
        ? { marginTop: height * 0.05, overflow: 'hidden' }
        : Platform.OS === 'ios'
        ? { marginTop: height * 0.08, overflow: 'hidden' }
        : { marginTop: height * 0.09, overflow: 'hidden' };

    const bodyStyle = props.isMain
        ? {
              zIndex: 1,
              paddingHorizontal: width * 0.05,
              overflow: 'hidden'
          }
        : {
              zIndex: -1,
              paddingTop: 75,
              paddingHorizontal: width * 0.05,
              overflow: 'hidden'
          };

    return (
        <View style={styles.container}>
            <Header style={{ ...props.headerStyles, ...headerStyle }}>
                {props.headerContent}
            </Header>
            <Body
                style={{ ...props.bodyStyle, ...bodyStyle }}
                accessibilityLabel={props.accessibilityLabel}
            >
                {props.bodyContent}
            </Body>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1
    }
});

export default PageContainer;
