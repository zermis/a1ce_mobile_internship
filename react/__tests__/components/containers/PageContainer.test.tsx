import PageContainer from '../../../src/containers/PageContainer';
import renderer from 'react-test-renderer';

it('renders page container correctly', () => {
    const tree = renderer.create(<PageContainer isMain={false} />).toJSON();
    expect(tree).toMatchSnapshot();
});
