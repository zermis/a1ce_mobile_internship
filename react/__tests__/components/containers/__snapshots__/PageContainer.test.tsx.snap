// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page container correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "height": "35%",
        "overflow": "hidden",
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4289279286,
            4281878109,
          ],
          "endPoint": Array [
            1,
            0.5,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            0.5,
          ],
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 80.03999999999999,
            "marginRight": 80.03999999999999,
            "marginTop": 106.72,
            "overflow": "hidden",
          }
        }
      />
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF8F8",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 1,
        "marginTop": -70,
        "overflow": "hidden",
        "paddingHorizontal": 37.5,
        "paddingTop": 75,
        "zIndex": -1,
      }
    }
  />
</View>
`;
